libevm/VM.cpp

```diff
@@ -87,15 +86,31 @@ void VM::onOperation()
 			m_newMemSize > m_mem.size() ? (m_newMemSize - m_mem.size()) / 32 : uint64_t(0),
 			m_runGas, m_io_gas, this, m_ext);
 }
+#if EVM_HACK_ON_OPERATION
+	#define onOperation()
+#endif
 
-void VM::checkStack(unsigned _removed, unsigned _added)
+//
+// set current SP to SP', adjust SP' per _removed and _added items
+//
+void VM::adjustStack(unsigned _removed, unsigned _added)
```

Comment:
I would suggest using `int` type here. 2 reasons: we need to do arithmetic in the function, so unsigned type is not good for this. Moreover, unsigned is sometimes pessimization because of _defined_ overflow.